pkg/infrastructure/azure/dns.go

```diff
@@ -0,0 +1,147 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/dns/armdns"
+	"github.com/openshift/installer/pkg/infrastructure/clusterapi"
+	"github.com/openshift/installer/pkg/types"
+)
+
+// Create DNS entries for azure.
+func createAzureDNSEntries(ctx context.Context, in clusterapi.InfraReadyInput) error {
+	clusterName := in.InfraID
+	private := in.InstallConfig.Config.Publish == types.InternalPublishingStrategy
+	resourceGroup := in.InstallConfig.Config.Azure.ResourceGroupName
+	zoneName := in.InstallConfig.Config.Azure.DefaultMachinePlatform.Zones
+	subscriptionId := in.InstallConfig.Azure.Credentials.SubscriptionID
+	apiExternalName := fmt.Sprintf("api.%s", clusterName)
+	apiExternalNameV6 := fmt.Sprintf("v6-api.%s", clusterName)
+	// TODO: set value from either manifests or getting from client.
+	var azureTags map[string]*string
+	var ipv4Elb string
+	var ipv4Ilb string
+	var ipv6Elb string
+	var ipv6Ilb string
```

Comment:
You can get the IP address for the internal LB from cluster spec. Here's an example from GCP:

https://github.com/openshift/installer/blob/6b4523dcdc6fa1cc5b676f6e0920e58d5a748a5d/pkg/infrastructure/gcp/clusterapi/clusterapi.go#L113-L124

Note that in this code, CAPG is creating the external LB so `.Spec.ControlPlaneEndpoint.Host` has the IP for the external LB, but in Azure we're using CAPZ to create the internal LB, so this field will include the IP for the internal LB. 

For the external LB, we will need to get that from the installer-created LB, like in #8115